packages/core/strapi/src/utils/transform-content-types-to-models.ts

```diff
@@ -0,0 +1,53 @@
+import { DatabaseConfig } from '@strapi/database';
+import { Schema, Attribute } from '@strapi/types';
+import { createId } from '@paralleldrive/cuid2';
+
+const transformAttribute = (attribute: Attribute.Any) => {
+  switch (attribute.type) {
+    case 'media': {
+      return {
+        type: 'relation',
+        relation: attribute.multiple === true ? 'morphMany' : 'morphOne',
+        target: 'plugin::upload.file',
+        morphBy: 'related',
+      };
+    }
+    default: {
+      return attribute;
+    }
+  }
+};
+
+export const transformContentTypesToModels = (
+  contentTypes: Pick<
+    Schema.ContentType,
+    'uid' | 'modelName' | 'collectionName' | 'attributes' | 'modelType'
+  >[]
+): DatabaseConfig['models'] => {
+  return contentTypes.map((contentType) => {
+    // Add document id to content types
+    // @ts-expect-error - `default` function is not typed into `Attribute`
+    // as it is not documented
+    const documentIdAttribute: Record<string, Attribute.Any> =
+      contentType.modelType === 'contentType'
+        ? { documentId: { type: 'string', default: createId } }
+        : {};
+
+    const model = {
+      ...contentType,
+      // reuse new model def
+      singularName: contentType.modelName,
+      tableName: contentType.collectionName!,
+      attributes: {
+        ...documentIdAttribute,
```

Comment:
how should we handle users creating a documentId attribute? Should we prevent this at this level by throwing an error?